### PR TITLE
colbert: add lint to colbert package and tests

### PR DESCRIPTION
**Pull request by nicoloboschi**

### CODE REVIEW

File: ragstack/colbert/colbert_embedding.py [hunk 1]
```diff
@@ -161,50 +158,54 @@ def encode_queries(
...
+        logging.info(
+            f"{len(tokens[0])} tokens in first query with query_maxlen {fixed_length}"
+        )
 
         self.checkpoint.query_tokenizer.query_maxlen = fixed_length
 
         # All query embeddings in the ColBERT documentation
         # this name, EQ or Q, maps the exact name in most colBERT papers
-        Q = self.checkpoint.queryFromText(
+        queriesQ = self.checkpoint.queryFromText(
```
Comment (cbornet) [inline ragstack/colbert/colbert_embedding.py:172]:
```suggestion
        queries = self.checkpoint.queryFromText(
```

File: ragstack/colbert/colbert_embedding.py [hunk 2]
```diff
@@ -161,50 +158,54 @@ def encode_queries(
...
             queries, bsize=bsize, to_cpu=True, full_length_search=full_length_search
         )
-
-        return Q
+        return queriesQ
```
Comment (cbornet) [inline ragstack/colbert/colbert_embedding.py:175]:
```suggestion
        return queries
```

File: ragstack/colbert/vector_store.py [hunk 3]
```diff
@@ -27,8 +22,28 @@ def put_document(self, document: str, metadata: dict):
...
+    body: str
+    rank: int
+    score: Number
+
+
+class ColBERTVectorStoreRetriever(ABC):
+    @abstractmethod
+    def close(self):
+        """Close the store."""
+        pass
```
Comment (cbornet) [inline ragstack/colbert/vector_store.py:42]:
```suggestion
```

File: ragstack/colbert/vector_store.py [hunk 4]
```diff
@@ -27,8 +22,28 @@ def put_document(self, document: str, metadata: dict):
...
+
+    @abstractmethod
+    def retrieve(
+        self, query: str, k: Optional[int], query_maxlen: Optional[int], **kwargs
+    ) -> List[Document]:
+        """Retrieve documents from the store"""
+        pass
```
Comment (cbornet) [inline ragstack/colbert/vector_store.py:49]:
```suggestion
```

**Comment by cbornet** [comment]
Weird. Colbert is installed but the test complains that it doesn't find the module 😕

File: ragstack/colbert/colbert_embedding.py (hunk 1 above)
Comment (zzzming) [reply ragstack/colbert/colbert_embedding.py:172]: Is there a way to align what's the variable in the paper and doc? I think the code should reflect what the paper describes. If Q cannot used, we should at least use queriesQ or something with a Q

**Review by zzzming — APPROVED** [review]

File: ragstack/colbert/colbert_embedding.py (hunk 1 above)
Comment (cbornet) [reply ragstack/colbert/colbert_embedding.py:172]: Then just `q` lowercase ?

**Pull request merged**